File: PhonePeAlgorithms.java
import java.util.*;
import java.io.*;

public class PhonePeAlgorithms {
    
    public static void main(String[] args) {
        nameSorted(new ArrayList<String>(Arrays.asList("Shailash", "Rahul", "Moumita")));
        cityNumber(1, 2, new int[]{1, 2, 3, 4, 5, 5});
    }

    public static ArrayList<String> nameSorted(ArrayList<String> nameList) {
        System.out.println("Original " + Arrays.toString(nameList.toArray()));
        
        HashMap<Integer, String> uniqueName = new HashMap<Integer, String>();
        for (int i = 0; i < nameList.size(); i++) {
            int count = (int) nameList.get(i).toLowerCase().chars().distinct().count();
            uniqueName.put(count, nameList.get(i));
        }

        Map<Integer, String> reverseSortedMap = new TreeMap<Integer, String>(Collections.reverseOrder());
        reverseSortedMap.putAll(uniqueName);

        ArrayList<String> newNameList = new ArrayList<String>(reverseSortedMap.values());
        Collections.sort(newNameList);

        System.out.println("nameSorted " + Arrays.toString(newNameList.toArray()));
        return newNameList;
    }

    public static int[] cityNumber(int K, int Y, int[] arr) {
        return arr;
    }
}

File: PhonePeAlgorithms.kt
class PhonePeAlgorithms {
    fun main() {
        val nameList = arrayListOf<String>("Shailash", "Rahul", "Moumita")
        nameSortedAlgorithm(nameList)
        cityNumberAlgorithm(4, 5, intArrayOf(16, 15, 14, 13, 12, 11, 10))
    }

    fun nameSortedAlgorithm(nameList: ArrayList<String>) {
        val uniqueName = mutableMapOf<Int, String>()
        nameList.forEach { uniqueName.set(it.toCharArray().distinct().count(), it) }
        println(uniqueName)
        val newNameList = uniqueName.toSortedMap().values.toList().reversed().sortedBy { it }
        println(newNameList)
    }

    fun cityNumberAlgorithm(k: Int, y: Int, arr: IntArray) {
        val list = MutableList<String>()
    }
}
